cmake/CheckLinuxKernel.cmake

```diff
@@ -0,0 +1,112 @@
+cmake_minimum_required(VERSION 3.10)
+
+# Retrieves the kernel version from /usr/include/linux/version.h and stores it in KERNEL_VER.
+# This should also correspond with the linux-libc version.
+# TODO: Consider other paths for linux/version.h?
+function(get_linux_libc_version_from_headers OUT_VERSION)
+    # macOS does not support these kinds of checks.
+    if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+        return()
+    endif()
+
+    set(LINUX_VERSION_HEADER "/usr/include/linux/version.h")
+
+    if (NOT EXISTS ${LINUX_VERSION_HEADER})
+        message(ERROR "linux/version.h not found! Ensure linux-libc-dev is installed.")
```

Comment:
Maybe this should not be an error? The fact we are not able to check kernel version just means we cannot use the targets that need certain version of it, but it does not mean the compilation of P4C should fail.